ser/train.py

```diff
@@ -2,19 +2,24 @@
 import torch
 from torch import optim
 from .model import Net
+from .model import MnistResNet
 import torch.nn.functional as F
+import torch.nn as nn
 
 from .data import train_data, val_data
 
 import typer
 
 import json
 
+import git
+
+repo = git.Repo(search_parent_directories=True)
+sha = repo.head.object.hexsha
```

Comment:
This is stylistic, but personally I would not put any code like this at the module level.

In this case it's fine, but it's a good habit to avoid anything but constants like the project directory definition at the module level.

The reason for this is that importing this module from another file will now have side effects!! This can lead to module imports failing or doing other weird stuff.